src/parser/mod.rs

```diff
@@ -7274,6 +7274,16 @@ impl<'a> Parser<'a> {
                 let placeholder = tok.to_string() + &ident.value;
                 Ok(Value::Placeholder(placeholder))
             }
+            tok @ Token::Minus | tok @ Token::Plus => {
```

Comment:
Normally I would expect that the `-` is included as part of the token itself  otherwise negative numbers wouldn't be able to be parsed at all anywhere

I wonder if the issue is that the of token being handled by create sequence needs to be expanded 🤔

Reply:
@alamb Not sure if I understand your point correctly. `token::Minus` might be a sign symbol for a number or a binary operation, and this cannot be determined when parsing it into a token.

Reply:
My primary concern is that this change seems to have much larger potential scope than simply supporting negative numbers for `CREATE SEQUENCE` -- it seems like it would change parsing of all values. 

However, I am clearly confused -- I added a test that shows  that `-10` is parsed as `-(10)` (aka a unary minus). See  https://github.com/sqlparser-rs/sqlparser-rs/pull/1421. That test still passes on this PR, so it must not have as much of an impact as I thought

Reply:
> My primary concern is that this change seems to have much larger potential scope than simply supporting negative numbers for CREATE SEQUENCE -- it seems like it would change parsing of all values.

It did affect the parsing of all values.

> However, I am clearly confused -- I added a test that shows that -10 is parsed as -(10) (aka a unary minus). See https://github.com/sqlparser-rs/sqlparser-rs/pull/1421. That test still passes on this PR, so it must not have as much of an impact as I thought

Yes, I also mentioned this in the comment https://github.com/sqlparser-rs/sqlparser-rs/pull/1419#discussion_r1751560914, the negative number will be parsed into an unary operation. So I'm wondering if we should keep it consistent by parsing it as an unary operation in PR as well.

Reply:
It seems when parsing expressions, the signed number literals are handled by the precedence logic when looking to parse a sub expr [here](https://github.com/sqlparser-rs/sqlparser-rs/blob/cb0c511b05f4acc5867dce6e8b00ec38884cd058/src/parser/mod.rs#L1158-L1172) is where we usually land. From what I could tell, we don't end up hitting the new code when parsing expressions due to calling `parse_prefix` beforehand - though I can't tell how well the change works with dialect overrides to the precedence logic so there might be some potential for surprising behavior. Also can't say for sure if there are other considerations other than when parsing subexpressions

The doc for [Token::Number](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/tokenizer.rs#L54-L55) at least explicitly says it contains unsigned number and since that's what `Value::Number` maps to, I think it might be reasonable to continue to always represent signed numbers as unary Exprs vs sometimes baked into the Value.

For this PR, issue seems that the clauses in [CREATE SEQUENCE](https://github.com/sqlparser-rs/sqlparser-rs/blob/cb0c511b05f4acc5867dce6e8b00ec38884cd058/src/parser/mod.rs#L11658) specifically are looking to parse number literals but the [parse_number_value](https://github.com/sqlparser-rs/sqlparser-rs/blob/cb0c511b05f4acc5867dce6e8b00ec38884cd058/src/parser/mod.rs#L7333) helper currently looks to parse exactly a `Value::Number`, maybe they can instead be updated to call `parse_expr`, expecially since the expected types on the enum are already [Expr](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/ast/mod.rs#L4712-L4715).

There are few similar calls to `parse_number_value()` I suspect they could have similar issue of rejecting signed numbers, so if we end up keeping the current behavior, a follow up might be to make it explicit what `parse_number_value` returns/rejects.